modules/apps/frontend-css/frontend-css-custom-properties-sample-web/src/main/resources/META-INF/resources/js/CssCustomPropertiesSample.js

```diff
@@ -0,0 +1,276 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+import React, {useState} from 'react';
+
+import '../css/main.scss';
+
+const CcpItem = ({children, sample, label}) => {
+	return (
+		<div className="ccp-item">
+			<span className={'ccp-sample' + (sample ? ' ' + sample : '')}>
+				{children && children}
+			</span>
+			{(label || sample) && (
+				<span className="ccp-label">{label || sample}</span>
+			)}
+		</div>
+	);
+};
+
+const CssCustomPropertiesSample = () => {
+	const [fade, setFade] = useState(false);
+	const [collapse, setCollapse] = useState(false);
+	const sampleText = 'The quick brown fox jumps over the lazy dog';
+
+	return (
+		<div className="ccp">
+			<div className="container">
+				<div className="row">
+					<div className="col">
+						<h1>CSS Custom Properties Sample</h1>
+					</div>
+				</div>
+
+				<div className="row">
+					<div className="ccp-group ccp-group-colors col-12">
+						<h2>Grays</h2>
+						<div className="ccp-items">
+							<CcpItem sample="bg-white" label="white" />
```

Comment:
Hey @carloslancha what are the advantages of using the array instead of the items?

Visually I think the second one looks tidier

<img width="583" alt="Screenshot 2020-08-24 at 12 42 46" src="https://user-images.githubusercontent.com/46778114/91036626-f94db380-e607-11ea-9077-560a2d24fbaf.png">
<img width="572" alt="Screenshot 2020-08-24 at 12 43 01" src="https://user-images.githubusercontent.com/46778114/91036647-04084880-e608-11ea-90c6-5765b80d21f8.png">

And in some case the value of `sample` needs to be different from `label` 

![image](https://user-images.githubusercontent.com/46778114/91036741-2a2de880-e608-11ea-8f44-8244a0e8a0f9.png)

Reply:
> Visually I think the second one looks tidier

I disagree. Also all those elements are expected to render the same component and behave in the same way, so you don't need to worry about adding or removing markup if the list changes, only elements to a list following the defined API, in this case just adding the string, or in the second case you mention:

```
[{
   label: 'border-radius',
   sample: 'rounded'
},
...
].map(item => (
   <CcpItem label={item.label} sample={item.sample} />
))}
```

Reply:
What @carloslancha is proposing is more idiomatic, since this is JS/React and not HTML. 

It's also more convenient if:
- The items in the list can change. It's easy to add, remove, edit, or even receive the list from the outside)
- The list is long. It's easier to locate items in a list than in similar lines of markup

One additional thing is that you don't have to inline the list as you did in your example. Since it's a constant, you'd simply declare it on top and just do `COLORS.map(...)`

Reply:
Yeah I know, I thought about this last night 🤓